src/modules/generators/fractals/fbm.rs

```diff
@@ -0,0 +1,163 @@
+// limitations under the License.
+
+use num_traits::Float;
+use math;
+use math::{Point2, Point3, Point4};
+use NoiseModule;
+use modules::Perlin;
+use super::util::*;
+
+/// Default noise seed for the Fbm noise module.
+pub const DEFAULT_FBM_SEED: usize = 0;
+/// Default number of octaves for the Fbm noise module.
+pub const DEFAULT_FBM_OCTAVE_COUNT: usize = 6;
+/// Default frequency for the Fbm noise module.
+pub const DEFAULT_FBM_FREQUENCY: f32 = 1.0;
+/// Default lacunarity for the Fbm noise module.
+pub const DEFAULT_FBM_LACUNARITY: f32 = 1.91803398875;
+/// Default persistence for the Fbm noise module.
+pub const DEFAULT_FBM_PERSISTENCE: f32 = 0.5;
+/// Maximum number of octaves for the Fbm noise module.
+pub const FBM_MAX_OCTAVES: usize = 32;
+
+/// Noise module that outputs classical fBm noise.
+#[derive(Clone, Debug)]
+pub struct Fbm<T> {
+    /// Seed.
+    pub seed: usize,
+
+    /// Total number of frequency octaves that generate the Fbm noise.
+    pub octaves: usize,
+
+    /// Frequency of the first octave.
+    pub frequency: T,
+
+    /// Gap between successive frequencies.
+    pub lacunarity: T,
+
+    /// Persistence of the value.
+    pub persistence: T,
+
+    sources: Vec<Perlin>,
+}
+
+impl<T: Float> Fbm<T> {
+    pub fn new() -> Fbm<T> {
+        Fbm {
+            seed: DEFAULT_FBM_SEED,
+            octaves: DEFAULT_FBM_OCTAVE_COUNT,
+            frequency: math::cast(DEFAULT_FBM_FREQUENCY),
+            lacunarity: math::cast(DEFAULT_FBM_LACUNARITY),
+            persistence: math::cast(DEFAULT_FBM_PERSISTENCE),
+            sources: build_sources(DEFAULT_FBM_SEED, DEFAULT_FBM_OCTAVE_COUNT),
+        }
+    }
+
+    pub fn set_seed(self, seed: usize) -> Fbm<T> {
+        if self.seed == seed {
+            return self;
+        }
+        Fbm {
+            seed: seed,
+            sources: build_sources(seed, self.octaves),
+            ..self
+        }
+    }
+
+    pub fn set_octaves(self, mut octaves: usize) -> Fbm<T> {
+        if self.octaves == octaves {
+            return self;
+        } else if octaves > FBM_MAX_OCTAVES {
+            octaves = FBM_MAX_OCTAVES;
+        }
+        Fbm {
+            octaves: octaves,
+            sources: build_sources(self.seed, octaves),
+            ..self
+        }
+    }
+
+    pub fn set_frequency(self, frequency: T) -> Fbm<T> {
+        Fbm { frequency: frequency, ..self }
+    }
+
+    pub fn set_lacunarity(self, lacunarity: T) -> Fbm<T> {
+        Fbm { lacunarity: lacunarity, ..self }
+    }
+
+    pub fn set_persistence(self, persistence: T) -> Fbm<T> {
+        Fbm { persistence: persistence, ..self }
+    }
+}
+
+/// 2-dimensional Fbm noise
+impl<T: Float> NoiseModule<Point2<T>> for Fbm<T> {
+    type Output = T;
+
+    fn get(&self, mut point: Point2<T>) -> T {
+        let mut value = T::zero();
+        let mut cur_persistence = T::one();
+
+        point = math::mul2(point, self.frequency);
+
+        for x in 0..self.octaves {
```

Comment:
There seems to be a commonality here between `Fbm` and `RidgedMulti` - is there a way to provide higher order modules for `frequency` and `octaves`?